deploy.sh

```diff
@@ -122,31 +132,27 @@ function retry_terraform {
 }
 
 function setup_terraform {
-  if [[ "$DISTRO" == *Ubuntu* ]]; then
-    PACKAGE_MANAGER="apt-get"
-  else
-    PACKAGE_MANAGER="yum"
+  TF_LATEST=$(curl -s https://api.github.com/repos/hashicorp/terraform/releases/latest | grep tag_name | cut -d'"' -f4)
+  if which terraform 2> /dev/null; then
+    TF=`which terraform 2> /dev/null`
   fi
 
-  if [ -f $TF ]; then
-    TF_VER=$($TF version | awk '{print $2}')
-    if [ "$TF_VER" == v"${TF_VERSION}" ]; then
-      log "Terraform already installed"
-      $TF version
-    fi
+  if [[ -f $TF && $("$TF" version | grep 'Terraform v0') == "Terraform ${TF_LATEST}" ]]; then
+    log "Terraform latest version already installed"
   else
-    log "Installing dependency packages"
-    $PACKAGE_MANAGER update -y > /dev/null 2>&1
-    $PACKAGE_MANAGER install -y git curl unzip > /dev/null 2>&1
+    if [[ "$PLATFORM" == *Linux* ]]; then
```

Comment:
We are checking the platform already in main function.. please make use of that.